Added color generation feature

aliyailina opened the pull request:
Added color generation in the same way it was added for images.

Also added some really minor refactor changes, such as typo fixes, remove useless usings and change readonly field names according to more convenient codestyle, hope it is OK.

Commit:
Added sample color

Commit:
Added support of Colors

Commit:
Minor refactoring

Remove useless code
Fix typos

Commit:
Fixed naming typos in .targets

Saratsin commented on src/Xamarin.iOS.ResourceDesigner/GenerateResourceDesignerTask.cs:
```diff
@@ -10,6 +10,9 @@ public class GenerateResourceDesignerTask : Task
         [Required]
         public ITaskItem[]? ImageAssets { get; set; }
+        
+        [Required]
+        public ITaskItem[]? ColorAssets { get; set; }
```
Inside of a csproj file color assets are represented by ImageAsset nodes. On a task level there is no need for such separation, we will just use ImageAssets property and filter items that we need

aliyailina replied:
Shouldn't it be renamed from ImageAssets to just Assets then? Yeah, I know that the resource tag is called ImageAsset, but it really looks like the wrong naming (as color assets was introduced in XCode 9, so I suppose there was no color assets when this tag was created) and "ImageAssets" name might be confusing for code readers. Then use Asset for anything ImageAsset tag-related (like AssetRawDto instead of ImageAsset...)?

aliyailina replied:
In this case we can continue use ColorAsset and ImageAsset naming, without renaming it to -Sets

Saratsin replied:
ImageAsset naming still remains ambiguous, since in csproj this node is like this. Probably you're right that it is a wrong naming in csproj files, but we will have it till if microsoft will rename this node

Saratsin replied:
I think we can do the following:
1. Call node `Assets`.  It will take as ITaskItem[] list of @(ImageAsset), since it is a legacy of Microsoft and Xamarin. But the name will clearly declare that they're assets in general.
2. In all occurences, rename `ImageAsset` into `ImageSet`. This will remove ambiguity with more general `ImageAsset` node in csproj files, declaring that these properties refer only to *.imageset items (iOS resources all image sets Contents.json files lays inside of *.imageset folders).
3. In all occurences, rename `ColorAsset` into `ColorSet`. This will have the same logic of naming as with `ImageSet`, but here declaring that these properties refer only to *.colorset items (iOS resources all color sets Contents.json files lays inside of *.colorset folders).

Saratsin commented on src/Xamarin.iOS.ResourceDesigner/GenerateResourceDesignerTask.cs:
```diff
@@ -19,9 +22,15 @@ public class GenerateResourceDesignerTask : Task
         public string? ProjectNamespace { get; set; }
+        
+        [Required]
+        public string? ColorAssetsTrimmingPrefixes { get; set; }
```
Inside of a csproj file color sets are represented by ImageAsset nodes. On a task level there is no need for such separation. However I understand that prefixes might be specific for each subtype (specifc for ImageSet-s, specific for ColorSet-s. Let's rename this property to ColorSetsTrimmingPrefixes. And, respectively, rename ImageAssetsTrimmingPrefixes into ImageSetsTrimmingPrefixes please

Saratsin commented on src/Xamarin.iOS.ResourceDesigner/GenerateResourceDesignerTask.cs:
```diff
@@ -19,9 +22,15 @@ public class GenerateResourceDesignerTask : Task
         public string? ImageAssetsTrimmingPrefixes { get; set; }
+        
+        [Required]
+        public string? ColorAssetFilenamesSeparatorChars { get; set; }
```
Inside of a csproj file color sets are represented by ImageAsset nodes. On a task level there is no need for such separation. However I understand that separator chars might be specific for each subtype (specifc for ImageSet-s, specific for ColorSet-s). And tbh, I think I've made too long name, "Filename" is redundant word there. Let's rename this property to ColorSetsSeparatorChars. And, respectively, rename ImageAssetFilenamesSeparatorChars into ImageSetsSeparatorChars please

Saratsin commented on src/Xamarin.iOS.ResourceDesigner/GenerateResourceDesignerTask.cs:
```diff
@@ -37,6 +46,13 @@ public override bool Execute()
                 FilenamesSeparatorChars = ImageAssetFilenamesSeparatorChars!.ToCharArray()
             };
+            
+            var colors = new ColorAssetRawDto
+            {
+                ColorAssetPaths = ColorAssets!.Select(asset => asset.GetMetadata("FullPath")).ToArray(),
+                TrimmingPrefixes = ColorAssetsTrimmingPrefixes!.Split('|'),
+                FilenamesSeparatorChars = ColorAssetFilenamesSeparatorChars!.ToCharArray()
+            };
```
I see two possible options, with changes that I wrote before:
1. Use ImageAssetsRawDto here as well (since node is ImageAsset), remove ColorAssetRawDto at all
2. Rename it to ColorSetsRawDto and rename property to ColorSetsPaths (maybe even just Paths, not specify ColorSets, since the class name says it already). Also to make the similar renaming with ImageAssetsRawDto - rename it to ImageSetsRawDto and rename its Paths property respectively

aliyailina replied:
What do you think about this:
Create base record for nodes (InterfaceDefinition, Image asset nodes, Color asset nodes, etc) which contains Paths property
The InterfaceDefinition is inheritor

Then there is base assets record which contains prefixes and separators properties + get-only property which identifies the set extension (colorset, imageset, etc).
Then in execute we create colors and images vars based on data.

Saratsin commented on src/Xamarin.iOS.ResourceDesigner/Resources/Resource.designer.mustache:
```diff
@@ -20,7 +20,7 @@ namespace {{{Namespace}}}
     {
         {{#ImageClass}}
-        public static partial class Image
+        public static partial class ImageAbuba
```
Revert please

aliyailina replied:
Sorry, it was for test purposes:)

Saratsin reviewed (changes requested):
Added some change requests

Commit:
Reverted test naming for generation template

Commit:
Apply renaming + README.md updated

Rename ImageAssets to just Assets and use ImageSet and ColorSet naming to separate between ImageAsset tag and image asset itself

Saratsin reviewed (approved):
LGTM
